pkg/signature/kms/hashivault/client.go

```diff
@@ -246,12 +246,17 @@ func (h *hashivaultClient) public() (crypto.PublicKey, error) {
 			return nil
 		},
 	)
+
+	item := h.keyCache.Get(cacheKey, ttlcache.WithLoader[string, crypto.PublicKey](loader))
 	if lerr != nil {
```

Comment:
Maybe this should go up

Reply:
Yeah before we added this up, but the chains controller pod was failing
Also as far as I understood, loader was called when we are assigning the value to `item` and hence if we add this error check up, at that time error is nil 
Adding it here, I think handles the error properly

Reply:
Good catch - If I'm understanding this correctly, lerr will always be nil in the previous code because it only gets set once the loader is invoked, which is in the `Get` call.

Reply:
yes exactly, and hence even if there was an error, it was not returned earlier and the `item`  would be nil which will indeed panic the code
Hence with the current patch, the error would be handled correctly